golang/cosmos/app/app.go

```diff
@@ -965,9 +977,14 @@ func (app *GaiaApp) InitChainer(ctx sdk.Context, req abci.RequestInitChain) abci
 		panic(err)
 	}
 
+	ensureBlockMaxBytes(app, ctx)
+
 	app.UpgradeKeeper.SetModuleVersionMap(ctx, app.mm.GetVersionMap())
 	res := app.mm.InitGenesis(ctx, app.appCodec, genesisState)
 
+	// Update tendermint consensus params with any changes made
+	res.ConsensusParams = app.BaseApp.GetConsensusParams(ctx)
```

Comment:
Hmm, this seems wrong - the `BaseApp` state should follow from the genesis, not vice-versa. Try doing the validation on `res.ConsensusParams` here, as suggested above, which ensures that the genesis file had an acceptable value, and then `BaseApp` should be initialized to that acceptable value.